compiler/coreSyn/CoreLint.hs

```diff
@@ -1227,20 +1227,16 @@ lintCoreAlt lookup_scrut scrut_ty scrut_weight alt_ty alt@(DataAlt con, args, rh
         -- We've already check
       lintL (tycon == dataConTyCon con) (mkBadConMsg tycon con)
     ; let { con_payload_ty = piResultTys (dataConRepType con) tycon_arg_tys
-          ; (_, pred, con_args, _) = dataConSig con
-          ; ex_tvs = dataConExTyVars con
-          -- Weights of the things that the data con brings into scope
-          -- MattP: TODO: Move this into DataCon
-          ; weights = (replicate (length ex_tvs + length pred) Omega)
-                        ++ map weightedWeight (con_args) }
+          ; ex_tvs_n = length (dataConExTyVars con)
+          ; weights = replicate ex_tvs_n Omega ++
+                      map weightedWeight (dataConRepArgTys con) }
```

Comment:
I think this deserves a comment to explain what happens.

Reply:
Done.